lib/storage/keyval/etcd.go

```diff
@@ -63,6 +64,8 @@ func NewETCD(cfg ETCDConfig) (*electingBackend, error) {
 		Backend: &backend{
 			Clock:    clock,
 			kvengine: engine,
+
+			cachedCompleteOperations: make(map[string]*storage.SiteOperation),
```

Comment:
I would do this as another Backend implementation, for example `CachingBackend` instead of sticking this into the regular backend. The CachingBackend would embed Backend and only reimplement GetSiteOperations method adding the caching functionality.

Reply:
I'm not sure I see the advantage to that approach.